docs/features/common_functional_options.md

```diff
@@ -31,6 +31,21 @@ func (g *TestLogConsumer) Accept(l Log) {
 }
 ```
 
+#### WithLogger
+
+- Not available until the next release of testcontainers-go <a href="https://github.com/testcontainers/testcontainers-go"><span class="tc-version">:material-tag: main</span></a>
+
+If you need to either pass logger to a container, you can use `testcontainers.WithLogger` for example:
+
+```golang
+func TestHandler(t *testing.T) {
+    logger := TestLogger(t)
```

Comment:
Let's add a short paragraph explaining what `TestLogger` is and does. It can go right before the code snippet.